tools/shared.py

```diff
@@ -2019,19 +2018,19 @@ def consider_archive(f):
 
     Building.read_link_inputs([x for x in files if not x.startswith('-')])
 
-    current_archive_group = None
-
     # Rescan a group of archives until we don't find any more objects to link.
     def scan_archive_group(group):
       loop_again = True
       logging.debug('starting archive group loop')
       while loop_again:
         loop_again = False
         for archive in group:
-          if consider_archive(archive):
+          if consider_archive(archive, False):
```

Comment:
nit: I'd call this as `consider_archive(archive, force_add=False)`, to make it clearer what's False

Reply:
done